usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateNoOverlappingStopTimeInTripBlockTest.java

```diff
@@ -1615,37 +1609,427 @@ void tripWithNullLastTimeShouldNotGenerateNotice() {
-                firstTripFirstStopTime, firstTripLastStopTime, secondTripFirstStopTime, secondTripLastStopTime,
-                thirdTripFirstStopTime, thirdTripLastStopTime, mockTimeUtils);
+    }
+
+    @Test
+    void overlappingTripsWithDifferentServiceIdShouldGenerateNoticeWhenComparingTripsReferencingBothFiles() {
+        // trips.txt
+        // | routeId | tripId | serviceId  | blockId |
+        // |---------|--------|----------- |---------|
+        // | 0       | 2      | a          | 7       |
+        // | 0       | 5      | b          | 7       |
+        // | 0       | 8      | c          | 7       |
+        // | 1       | 9      | d          | 7       |
+
+        // stop_times.txt
+        // | tripId | arrivalTime | departureTime | stopId | stopSequence  |
+        // |--------|-------------|---------------|--------|---------------|
+        // | 2      | 05:00       | 05:00         |  99    |  8            |
+        // | 2      | 06:00       | 06:30         | 100    |  9            |
+        // | 5      | 07:00       | 07:40         | 101    | 10            |
+        // | 5      | 08:20       | 10:00         | 102    | 11            |
+        // | 8      | 11:03       | 11:15         | 103    | 12            |
+        // | 8      | 11:16       | 11:20         | 104    | 13            |
+        // | 9      | 11:10       | 11:28         | 105    | 14            |
+        // | 9      | 11:34       | 12:45         | 106    | 15            |
+
+        // trips.txt + stop_times.txt
+        // | routeId | tripId | serviceId | blockId | first stop time | last stop time |
+        // |---------|--------|-----------|---------|-----------------|----------------|
+        // | 0       | 2      | a         | 7       | 05:00           | 06:30          |
+        // | 0       | 5      | b         | 7       | 07:00           | 10:00          |
+        // | 0       | 8      | c         | 7       | 11:03           | 11:20          |
+        // | 1       | 9      | d         | 7       | 11:10           | 12:45          |
+
+        // calendar.txt:
+        // | serviceId | monday | tuesday | wednesday | thursday | friday | saturday | sunday | start_date | end_date |
+        // |-----------|--------|---------|-----------|----------|--------|----------|--------|------------|----------|
+        // | d         | 1      | 1       | 1         | 1        | 1      | 1        | 1      | 20200815   | 20200831 |
+
+        // calendar_dates.txt:
+        // | serviceId | date     | exception_type |
+        // |-----------|----------|----------------|
+        // | a         | 20200723 | 1              |
+        // | a         | 20200801 | 1              |
+        // | b         | 20200904 | 0              |
+        // | c         | 20201008 | 1              |
+
+        // Here services c and d overlap
+        // - service c is active only on 2020-01-08
```

Comment:
The above table says that service `c` is active on Oct. 8, 2020. But that still wouldn't overlap service `d`, which only runs from Aug 15 to Aug 31.